compiler/rustc_middle/src/mir/consts.rs

```diff
@@ -319,15 +319,13 @@ impl<'tcx> Const<'tcx> {
     ) -> Result<ConstValue<'tcx>, ErrorHandled> {
         match self {
             Const::Ty(_, c) => {
-                // We want to consistently have a "clean" value for type system constants (i.e., no
-                // data hidden in the padding), so we always go through a valtree here.
-                match c.eval_valtree(tcx, param_env, span) {
```

Comment:
Evaluating here is pointless as MIR is normalized on construction so the only aliases we should encounter will be rigid aliases that cannot be furthur normalized. Monomorphization will also immediately normalize any instantiated rigid aliases turning them into `ConstKind::Value` so this should also not do anything during codegen